app/src/pages/edit.tsx

```diff
@@ -0,0 +1,170 @@
+    };
+
+    const onChange = (field: string, value: string | boolean | Date | null | Dayjs) => {
+        if (formValues) {
+            setFormValues({
+                ...formValues,
+                [field]: value
+            });
+
+            if (["name", "surName", "phone", "email"].includes(field)) {
+                setFormErrors({
+                    [field]: value ? "" : REQUIRED_FIELD
+                })
+            }
+        }
+    };
+
+    const onSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+        try {
+            event.preventDefault();
+
+            if (user && formValues && !saveDisabled) {
+                const result = {
+                    ...formValues,
+                    userId: user.id
+                };
+
+                if (result["birthday"] && typeof result["birthday"] !== "string") {
+                    result["birthday"] = (result["birthday"] as dayjs.Dayjs).format("YYYY-MM-DD");
+                }
+
+            profile.editInfo(result, setLoading, setShowAlert)
+
+            } else {
+                throw new Error("Нет пользователя");
+            }
+        } catch (error) {
+            setLoading(false);
+            catchErrors.catch("Произошла ошибка при изменении информации о пользователе: " + error);
+        }
+    };
+
+    return <Paper className={"edit-container"}>
+        <Tabs
+            orientation="vertical"
+            value={tab}
+            onChange={onChangeTab}
+            aria-label="Edit tabs"
+            className={"edit-container__tabs"}
+        >
+            <Tab label="Основное" id="main" aria-controls="main" disabled={loading} className={"edit-container__tab-name"} />
+            <Tab label="Контакты" id="contacts" aria-controls="contacts" disabled={loading} className={"edit-container__tab-name"} />
+        </Tabs>
+
+        <div className={"edit-container__module"}>
+            {user && userDetail && formValues
+                ? <Box component="form" noValidate onSubmit={onSubmit}>
+                    <EditTabsModule tab={tab} formValues={formValues} formErrors={formErrors} onChange={onChange} />
+
+                    <LoadingButton
+                        fullWidth
+                        type="submit"
+                        variant="contained"
+                        className={"edit-container__module__loading-button"}
+                        loading={loading}
+                        disabled={saveDisabled}
+                    >
+                        Сохранить
+                    </LoadingButton>
+
+                    {showAlert
+                        ? <AlertComponent show={showAlert}>
+                            <><b>Изменения сохранены</b> - новые данные будут отражены на Вашей странице.</>
+                        </AlertComponent>
+                        : null
+                    }
+                </Box>
+                : <div className={"edit-container__module__spinner"}><CircularProgress /></div>
```

Comment:
we already have a common Spinner component - you can find him in app/src/components/common/Spinner